### DISCUSSION TITLE
About background color

### DISCUSSION THREAD

**Discussion by HolyZhit** [koreader/koreader#9412] [Q&A]
Hi!
As an Android user I am amazed at the performance of the koreader, but I am not used to its ink screen. I would like to know how to change the background of the reading to the parchment style or other color, can anyone please let me know, thanks.

**Comment by poire-z** [comment]
You can have that, but only with crengine documents (EPUB, HTML, FB2), not with PDF/DJVU. From https://github.com/koreader/koreader/pull/6188 :
> Add support for 2 settings one can set manually in reader.settings.reader.lua:
```
["cre_background_color"] = "0xccffcc",
["cre_background_image"] = "/path/to/bg_paper.jpg",
```

**Reply by HolyZhit** [reply]
Thank you*^____^*

**Reply by lweendela** [reply]
Thanks

**Reply by ghost** [reply]
> You can have that, but only with crengine documents (EPUB, HTML, FB2), not with PDF/DJVU. From #6188 :
> 
> > Add support for 2 settings one can set manually in reader.settings.reader.lua:
> 
> ```
> ["cre_background_color"] = "0xccffcc",
> ["cre_background_image"] = "/path/to/bg_paper.jpg",
> ```

@poire-z Sadly I'm not certain where to add the line in `settings.reader.lua` — I pasted the first line under `[style_tweaks]`, but doesn't seem to have any effect. I've restarted Koreader (2023.03 from Github) and tried with several EPUB files.

```
    [style_tweaks] = {
        [cre_background_color] = 0xccffcc,
        [font_family_all_inherit] = true,
        [font_size_most_reset] = true,
        [footnote-inpage_epub_smaller] = true,
        [footnote-inpage_fb2] = true,
        [line_break_cre_loose] = true,
    },
```

---

To confirm, `cre_background_color` and `cre_background_image` are independent of each other (I'm not planning to set a background image at the time)?

**Reply by Frenzie** [reply]
Not in style_tweaks, on the same level as style tweaks, but more important without the quotes it'll never have any effect because it'll error out on the reference to an undefined variable.

**Reply by ghost** [reply]
> Not in style_tweaks, on the same level as style tweaks, but more important without the quotes it'll never have any effect because it'll error out on the reference to an undefined variable.

The quotes `"` were lost due to copying with [wl-clipboard](https://github.com/bugaevc/wl-clipboard), sorry.

Thank you, works now (I added to the second line in the config file).

**Reply by shuvashish76** [reply]
> Add support for 2 settings one can set manually in `reader.settings.reader.lua`

I'm on F-Droid version. What's the location of this file? I can't find it in `/storage/emulated/0/koreader` directory.
`reader.settings.reader.lua` now changed to `settings.reader.lua` or it's a different file?

**Reply by Frenzie** [reply]
That was a typo. It's `settings.reader.lua.`

**Reply by shuvashish76** [reply]
It works for most epub files but when trying gutenberg epub files (for example [1](https://www.gutenberg.org/ebooks/12), [2](https://www.gutenberg.org/ebooks/11), [3](https://www.gutenberg.org/ebooks/97)) it never works. I need to turn off "Embedded Style" for gutenberg files to make them work. For other epub files it doesn't matter if I turn off or on the "Embedded Style", it always work.
What's the reason for this issue?

**Reply by Frenzie** [reply]
I'd assume it defines its own background to be painted on top of the crengine background. I think there are a few style tweaks to force various background behaviors without disabling embedded styles completely.

**Reply by poire-z** [reply]
We have a styletweak that is a bit radical and should solve your cases without killing all embedded styles:
https://github.com/koreader/koreader/blob/97a7ebefb997d31336e8ed1f64fe5134151ded50/frontend/ui/data/css_tweaks.lua#L795-L806
You probably could make another less radical one to just kill background-color set on `<body>`:
```css
body {
  background-color: transparent !important; 
  background-image: !important; 
}
```

**Reply by shuvashish76** [reply]
How to I add them to the KOReader?

**Reply by benoit-pierre** [reply]
By adding some css files to `koreader/styletweaks`, see https://github.com/koreader/koreader/pull/3944#issuecomment-389069259.

**Reply by shuvashish76** [reply]
> You probably could make another less radical one to just kill background-color set on `<body>`:

Thanks now it works for gutenberg files.👍

**Reply by shuvashish76** [reply]
When adding that as "Show in action list" it doesn't work. Actually none of the _possible_ options from "Style tweaks" work as "Show in action list".
They should appear under `Taps and gestures → Gesture manager → ANY GESTURE → General` right? or I misunderstood the Show in action list feature?

**Reply by hius07** [reply]
Under "Reflowable documents".

**Comment by dlgoodr** [comment]
Ooh, how about night mode? does that have a setting, too?

**Reply by lweendela** [reply]
Yes. Its there on setting. Just select the check box for night mode
![Screenshot_20230225-080534-1](https://user-images.githubusercontent.com/51630382/221341593-df838486-2e39-4200-ae00-3d480d273a77.jpg)

**Reply by dlgoodr** [reply]
Yes, I mean, is there a setting to change the color of the background in night mode to something other than black?

**Reply by NiLuJe** [reply]
No, it's designed to mimic eInk's behavior, which is a plain inversion ;).

**Reply by dlgoodr** [reply]
Fair, thanks.

**Comment by user-1-2-3-4-5** [comment]
`["cre_background_color"] = "0xADD8E6",` uses a light blue color, which I more or less prefer.

It may be necessary to set 'embedded style' to `off` for some ebooks.

**Comment by shuvashish76** [comment]
User patch available at https://github.com/Euphoriyy/KOReader.patches?tab=readme-ov-file#-2-ui-background-colorlua
> A toggle to change the page background color. (epub, html, fb2, txt...)

Edit: Now available as a single plugin: https://github.com/Euphoriyy/appearance.koplugin

**Reply by shuvashish76** [reply]
> A toggle to change the page background color. (pdf, djvu, cbz...)

Now supports fixed layout documents as well.
